snap/README.md

```diff
@@ -0,0 +1,85 @@
+# EdgeX USB Camera Device Service Snap
+[![edgex-device-usb-camera](https://snapcraft.io/edgex-device-usb-camera/badge.svg)](https://snapcraft.io/edgex-device-usb-camera)
+
+This directory contains the snap packaging of the EdgeX USB Camera device service.
+
+The snap is built automatically and published on the Snap Store as [edgex-device-usb-camera].
+
+For usage instructions, please refer to Device Camera section in [Getting Started using Snaps][docs].
+
+## Build from source
+Execute the following command from the top-level directory of this repo:
+```
+snapcraft
+```
+
+This will create a snap package file with `.snap` extension. It can be installed locally by setting the `--dangerous` flag:
+```bash
+sudo snap install --dangerous <snap-file>
+```
+
+The [snapcraft overview](https://snapcraft.io/docs/snapcraft-overview) provides additional details.
+
+### Obtain a Secret Store token
+<!-- The `edgex-secretstore-token` snap slot makes it possible to automatically receive a token from a locally installed platform snap.-->
+
+If the snap is built and installed locally, the interface will not auto-connect. You can check the status of the connections by running the `snap connections edgex-device-usb-camera` command.
+
+Notes:
+- The auto connection will not happen right now because the snap publisher isn't same as the `edgexfoundry` platrform snap (i.e. Canonical).
```

Comment:
Is it possible to run this snap in non-secure mode?

Reply:
yes, it is possible, I have tested its streaming function in no-secure mode:
```
snap set edgex-device-usb-camera app-options=true
snap set edgex-device-usb-camera config.edgex-security-secret-store=false
```